service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -137,6 +137,9 @@ where
         use rusoto_core::region;
         use rusoto_core::{{Client, RusotoFuture}};
 
+        #[allow(unused_imports)]
```

Comment:
I'd love to keep the services that don't need this trimmer in their imports. How about we remove the changes in this file and make something like the following changes in `query.rs`?

```rust
fn generate_prelude(&self, writer: &mut FileWriter, service: &Service) -> IoResult {
        let service_specific_prelude = match service.name().to_ascii_lowercase().as_ref() {
            "iam" => "use rusoto_core::signature::decode_uri;",
            _ => "",
        };
        let prelude = format!("use std::str::FromStr;
            use xml::EventReader;
            use xml::reader::ParserConfig;
            use rusoto_core::param::{{Params, ServiceParams}};
            use rusoto_core::signature::SignedRequest;
            use xml::reader::XmlEvent;
            use rusoto_core::xmlutil::{{Next, Peek, XmlParseError, XmlResponse}};
            use rusoto_core::xmlutil::{{characters, end_element, find_start_element, start_element, skip_tree, peek_at_name}};
            use rusoto_core::xmlerror::*;
            use serde_urlencoded;
            {}

            enum DeserializerNext {{
                Close,
                Skip,
                Element(String),
        }}", service_specific_prelude);

        writeln!(writer, "{}", prelude)
    }
```

This way only IAM will get the import it needs. 👍

Reply:
Or alternatively call the function via the fully-qualified name, e.g. `rusoto_core::signature::decode_uri(...)`.